tests/js/adapter-tests.ts

```diff
@@ -150,6 +150,30 @@ describe('Adapter', () => {
         tmpDir.removeCallback();
     });
 
+    it("predicate functions", async () => {
```

Comment:
This should also test that non-matching paths aren't watched (currently the test would pass if the adapter unconditionally watched all realms and ignored the predicate).